### PR TITLE
Add md5 to Biblio and Biblio contributor

### PR DESCRIPTION
**Pull request by amitaibu**
#43

### CODE REVIEW

File: plugins/biblio_style/bibtex/bibtex.inc [hunk 1]
```diff
@@ -5,9 +5,12 @@ $plugin = array(
   'description' => t('BibTeX style.'),
   'class' => 'BiblioStyleBibtex',
   // Indicates this style plugin can import entries.
-  'import' => TRUE,
+  'import' => array(
```
Comment (amitaibu) [inline plugins/biblio_style/bibtex/bibtex.inc]: @NofarGizra this should be removed please.

File: plugins/biblio_style/abstract.inc [hunk 2]
```diff
@@ -58,4 +58,27 @@ abstract class BiblioStyleBase implements BiblioStyleInterface {
   public function settingsFormValidate() {
 
   }
+
+  public function isDuplicate(Biblio $biblio) {
```
Comment (amitaibu) [inline plugins/biblio_style/abstract.inc]: If a duplicate is found, return the loaded biblio, so later we can show it to the user.

Comment (amitaibu) [inline plugins/biblio_style/abstract.inc]: > return the loaded biblio

Actually enough just the Biblio ID.

File: plugins/biblio_style/abstract.inc [hunk 3]
```diff
@@ -59,6 +59,112 @@ abstract class BiblioStyleBase implements BiblioStyleInterface {
...
+   * The md5 is later used to determine whether or not two Biblio objects are the
+   * same and prevent duplications.
+   *
+   * @param $biblio
+   *  Biblio object.
+   *
+   * @return
+   *  MD5 string that represents the given biblio.
+   */
+  public static function generateBiblioMd5($biblio) {
```
Comment (NofarGizra) [inline plugins/biblio_style/abstract.inc]: `(Biblio $biblio)`

File: plugins/biblio_style/bibtex/bibtex.inc [hunk 4]
```diff
@@ -5,9 +5,9 @@ $plugin = array(
   'description' => t('BibTeX style.'),
   'class' => 'BiblioStyleBibtex',
   // Indicates this style plugin can import entries.
-  'import' => TRUE,
+  'import' => 'TRUE',
```
Comment (amitaibu) [inline plugins/biblio_style/bibtex/bibtex.inc]: `'TRUE'` =>  `TRUE`

File: plugins/biblio_style/bibtex/BiblioStyleBibtex.class.php [hunk 5]
```diff
@@ -47,47 +46,18 @@ public function import($data) {
...
-      }
-
-      if (!empty($entry['keywords'])) {
-        if (strpos($entry['keywords'], ';')) {
-          $entry['keywords'] = str_replace(';', ',', $entry['keywords']);
+      if (empty($options['allow_duplicate'])) {
+        // Check if this is a unique Biblio.
+        if ($duplicate_id = $this->isDuplicate($biblio)) {
+          // Not unique, display message to the user.
+          drupal_set_message(t('Biblio "@title" already imported, view it <a href="@url">here</a>.', array('@title' => $biblio->title, '@url' => url('biblio/' . $duplicate_id))));
```
Comment (amitaibu) [inline plugins/biblio_style/bibtex/BiblioStyleBibtex.class.php:53]: @RoySegall change this return to be `return array('duplicate' ...`

**Comment by amitaibu** [comment]
Merged. Test should be done on another branch.

**Pull request closed without merging**